### ISSUE TITLE
vagrant ssh vm -c '...' output contains CRLF

### ISSUE THREAD

**Issue by roman-mueller** [mitchellh/vagrant#7146]
### Vagrant version

$ vagrant --version
Vagrant 1.8.1
### Host operating system

CentOS 6.7
### Guest operating system

CentOS 6.7 / LXC
### Vagrantfile

That file contains nothing special, it was generated via `vagrant init visibilityspots/centos-6.x-puppet-4.x` 

``` ruby
Vagrant.configure(2) do |config|
  config.vm.box = "visibilityspots/centos-6.x-puppet-4.x"
end
```
### Debug output

https://gist.github.com/roman-mueller/2de792b7b2cdbda297ce
### Expected behavior

The output of the `vagrant ssh default -c '...'` should not contain Windows style line breaks.
This is rather strange as this is running completely on Linux systems, nothing should convert line breaks like that.
### Actual behavior

The output did contain Windows style line breaks.
### Steps to reproduce
1. Run: `vagrant ssh default -c '/bin/ls -l /etc' > ls.txt` 
2. The file contains CRLFs:

```
$ file ls.txt
ls.txt: ASCII text, with CRLF line terminators
$ cat -A ls.txt
total 1132^M$
-rw-r--r--  1 root root      12 Feb 10 00:21 adjtime^M$
-rw-r--r--  1 root root    1512 Jan 12  2010 aliases^M$
(...)
```

**Comment by shirshendu** [comment]
I did some digging on this, and it seems like it is not really Vagrant-specific behaviour, but rather a quirk of how the `ssh` command works.
When invoked with the `-t` parameter, i.e. through a pseudo TTY, you get CRLF endings in the output.
Vagrant's ssh execution uses this parameter, because the user might want to interact with the command while it executes.
However, if you execute `vagrant ssh` without a TTY (say, from a cron job), the output automatically reverts to non-CRLF endings. I tested it using this command:
```bash
setsid sh -c "ssh vagrant@redacted -p 22 -o Compression=yes \
-o DSAAuthentication=yes -o LogLevel=FATAL -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
-o IdentitiesOnly=yes -i /home/vagrant-path/.vagrant/machines/default/libvirt/private_key \
-t \"bash -l -c '/bin/ls -l /etc'\"" < /dev/null > ls.txt
```
